### PR TITLE
debian setup: replace policykit-1 with pkexec

### PR DESCRIPTION
**Pull request by tomasmatus**
* [x] image-refresh debian-testing

Fixes: #6611

```
Package policykit-1 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  polkitd pkexec
```

### CODE REVIEW

File: images/scripts/debian.setup [hunk 1]
```diff
@@ -28,7 +28,8 @@ libvirt-dbus \
 network-manager \
 packagekit \
 pcp \
-policykit-1 \
+polkitd \
+pkexec \
```
Comment (martinpitt) [inline images/scripts/debian.setup:32]: Thanks! However, let's make this more precise:
```
# policykit-1 was split into multiple packages; keep old name for Debian 11 and Ubuntu
Recommends: sudo | pkexec | policykit-1
```
I.e.  Please drop the `polkitd` package. It's not a COCKPIT_DEPS, neither conceptually nor technically. Enough daemons (like udisks2) already pull it in.

The split package [exists in Debian stable](https://packages.debian.org/search?keywords=pkexec) already. However, [ubuntu-2204](https://launchpad.net/ubuntu/+source/policykit-1/0.105-26ubuntu1.3) does not yet have that.

So this needs a special case for ubuntu-2204, similar to libblockdev-mdraid2 before.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Thanks! Please adjust, build the images, and add the "Fixes #" to the commit message.

File: images/scripts/debian.setup (hunk 1 above)
Comment (tomasmatus) [reply images/scripts/debian.setup]: Thanks for pointing it out, I forgot about ubuntu :)

**Comment by martinpitt** [comment]
Good enough, thanks! Please turn the crank :gear:!

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-debian-testing-ab722ff1-20240715-085911/log.html

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-debian-testing-ab722ff1-20240715-094609/log.html

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-debian-testing-ab722ff1-20240715-111637/log.html

**Comment by tomasmatus** [comment]
tested locally, image build probably fails because of:
```
+ /var/lib/testvm/zero-disk.setup
dd: writing to '/root/junk': No space left on device
13365281+0 records in
13365280+0 records out
6843023360 bytes (6.8 GB, 6.4 GiB) copied, 12.044 s, 568 MB/s
image-create: Unable to reach machine debian-testing-127.0.0.2-2201 via ssh: 127.0.0.2:2201
```

**Comment by martinpitt** [comment]
Steps:
```
./image-create -v debian-testing
./vm-run /var/tmp/debian-testing*qcow2
ssh -vv c
```
This hangs for a long time indeed, while `ssh localhost` in the terminal works. `nft list tables` is empty, and `iptables -L` is mostly empty (just libvirt rules).

The reason is simply that there are *no* configured ethernet devices:
```
# ip -c a
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host noprefixroute 
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:54:00:12:34:56 brd ff:ff:ff:ff:ff:ff
    altname enp0s14
    altname ens14
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether 52:54:01:00:00:02 brd ff:ff:ff:ff:ff:ff
    altname enp0s15
    altname ens15

# networkctl 
IDX LINK TYPE     OPERATIONAL SETUP    
  1 lo   loopback carrier     unmanaged
  2 eth0 ether    carrier     unmanaged
  3 eth1 ether    carrier     unmanaged

# nmcli d
DEVICE  TYPE      STATE                   CONNECTION 
lo      loopback  connected (externally)  lo         
eth0    ethernet  disconnected            --         
eth1    ethernet  disconnected            -- 
```

**Comment by martinpitt** [comment]
Comparing to the current debian-testing image:
```
# nmcli d
DEVICE  TYPE      STATE                   CONNECTION   
eth0    ethernet  connected               netplan-eth0 
lo      loopback  connected (externally)  lo           
virbr0  bridge    connected (externally)  virbr0       
eth1    ethernet  disconnected            --           
```

/etc/netplan/50-cloud-init.yaml didn't change, but there's no /run/NetworkManager/system-connections/netplan-eth0.nmconnection . Running `netplan generate` shows why:

```
/etc/netplan/50-cloud-init.yaml:10:1: Invalid YAML: aliases are not supported:


^
```

And indeed that file is broken, it has an unterminated string constant:
```yaml
network:
  version: 2
  ethernets:
    eth0:
      match:
        macaddress: "52:54:00:12:34:56"
      dhcp4: true
      dhcp6: true
      set-name: "eth0
```

Fixing that and rebooting fixes the networking. This could plausibly coincide with [cloud-init 24.2-1](https://tracker.debian.org/news/1541661/accepted-cloud-init-242-1-source-into-unstable/).

As the initial boot is fine, I am 95% sure that this is fallout from our seddery:

    sed -i 's/ens[^[:space:]:]*/eth0/' /etc/network/interfaces /etc/network/interfaces.d/* /etc/netplan/*.yaml || true

@tomasmatus want to take over again, or shall I continue?

File: images/scripts/debian.setup [hunk 2]
```diff
@@ -318,13 +318,6 @@ systemctl disable apt-daily.timer apt-daily-upgrade.timer fstrim.timer man-db.ti
 echo "kernel.core_pattern=|/lib/systemd/systemd-coredump %P %u %g %s %t %c %e" > /etc/sysctl.d/50-coredump.conf
 printf 'DefaultLimitCORE=infinity\n' >> /etc/systemd/system.conf
 
-# disable udev network names, our tests expect the kernel schema
```
Comment (martinpitt) [inline images/scripts/debian.setup:315]: :clap: I'm glad this gets dropped at last. I'm happy to help with any test fallout.

Please add a detailed commit message for this, in the spirit of "our tests don't need that any more" and "let's test the *actual* OS".

**Comment by cockpituous** [comment]
image-refresh debian-testing done: https://github.com/cockpit-project/bots/commits/image-refresh-debian-testing-20240716-113153

**Comment by cockpituous** [comment]
Success. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-debian-testing-40fd77bc-20240716-111921/log.html

**Comment by tomasmatus** [comment]
failed tests might be related to #5016

**Comment by martinpitt** [comment]
@tomasmatus That's exactly it -- please reopen the naughty issue and  revive the pattern (it's still in git history). Nice progress!

**Review by martinpitt — APPROVED** [review]
Thanks! The test queue is very full, so this will take a bit still. But that Should Go Green™ :crossed_fingers:

**Pull request merged**